mlflow/models/python_api.py

```diff
@@ -146,20 +160,49 @@ def predict(
         # Run prediction with additional pip dependencies
         mlflow.models.predict(
             model_uri=f"runs:/{run_id}/model",
-            input_data='{"x": 1, "y": 2}',
+            input_data={"x": 1, "y": 2},
             content_type="json",
             pip_requirements_override=["scikit-learn==0.23.2"],
         )
 
     """
+    # to avoid circular imports
+    from mlflow.pyfunc import _PREBUILD_ENV_ROOT_LOCATION
+
     if content_type not in [_CONTENT_TYPE_JSON, _CONTENT_TYPE_CSV]:
         raise MlflowException.invalid_parameter_value(
             f"Content type must be one of {_CONTENT_TYPE_JSON} or {_CONTENT_TYPE_CSV}."
         )
 
+    spark_session = None
+    if is_in_databricks_runtime():
+        try:
+            # if in Databricks, fetch the current active session
+            from databricks.sdk.runtime import spark
+
+            spark_session = spark
+        except Exception:
```

Comment:
do we need to rely on databricks SDK to get the active spark session?

Reply:
Good point! This should be able to run locally as well. Updated!